torch/csrc/autograd/record_function.cpp

```diff
@@ -185,6 +198,13 @@ void RecordFunction::processCallbacks() {
   }
 }
 
+void RecordFunction::setThreadId() {
+  TORCH_INTERNAL_ASSERT(
+      torch::autograd::profiler::thread_id != 0,
+      "Can only call RecordFunction::setThreadId after RecordFunction::before has been run in this thread.");
+  threadId_ = torch::autograd::profiler::thread_id;
+}
+
 RecordFunction::~RecordFunction() {
```

Comment:
Hmm, not sure why clang-tidy is showing up here, as the destructor hasn't changed. Although it calls `end()` and I did change `AT_ASSERT` to `TORCH_INTERNAL_ASSERT`

Reply:
Let's catch the exception here and log it instead.